### ISSUE TITLE
Is there support for Set or at least NSSet in Bond 4?

### ISSUE THREAD

**Issue by juandent** [SwiftBond/Bond#153]
if no support currently available, what would you suggest as a workaround?

CoreData uses NSSet and NSOrderedSet, how do we interact with these if there is no ObservableSet?

One idea (don't know about performance) would be to create an ObservablArray from the set's contents and interact with that instead of the set itself! ??

Any ideas?
Thank you and ever grateful for this superb framework!!

Juan Dent

**Comment by srdanrasic** [comment]
Yes, ObservableSet and ObservableDictionary might be interesting features. I had something like this in mind with Bond 4 but didn't have enough time to implement it. I'll put it in the ToDo list.

Regarding CoreData, I'm not sure what kind of integration do you have in mind? You could use KVO to bridge scalar values to Bond. Sets would not work. Only way would be to convert them manually to ObservableArrays. Having less than a 100, maybe even 1000, objects in a Set would probably not impact performance in a significant way. CoreData (and even Realm) integration might be another ToDo item.

```swift
class Person: NSManagedObject {
  @NSManaged var name: String
}

extension Person {
  class ObservablePerson {
    let name: Observable<String>
    
    init(person: Person) {
      name = Observable(object: person, keyPath: "name")
    }
  }
  
  func observableVariant() -> Person.ObservablePerson {
    return ObservablePerson(person: self)
  }
}

let p = Person(...).observableVariant()
p.name.bindTo(nameLabel)
```

**Comment by juandent** [comment]
Hi!

O tried this idea but it does not compile

Let name : observable<String>. 
Cannot be initialized with 

> name = Observable(object: person, keyPath: "name")


??
Regards 
Juan dent

**Comment by srdanrasic** [comment]
Do you have these two imports?

```swift
import Bond
import CoreData
```

It compiles fine for me...

**Comment by juandent** [comment]
Yes now it works!! 

Thanks!

Juan
